runtime/lua/vim/lsp/completion.lua

```diff
@@ -706,6 +851,37 @@ local function enable_completions(client_id, bufnr, opts)
         end
       end,
     })
+
+    if vim.o.completeopt:find('popup') then
+      api.nvim_create_autocmd('Completechanged', {
+        group = group,
+        buffer = bufnr,
+        callback = function(args)
+          local completed_item = api.nvim_get_vvar('event').completed_item or {}
+          if (completed_item.info or '') ~= '' then
+            local data = vim.fn.complete_info({ 'selected' })
+            update_popup_window(data.preview_winid, data.preview_bufnr)
+            return
+          end
+
+          if
+            #lsp.get_clients({
+              id = vim.tbl_get(completed_item, 'user_data', 'nvim', 'lsp', 'client_id'),
+              methods = ms.completionItem_resolve,
+            }) == 0
```

Comment:
some LSP servers aren't giving us documentation, because we don't tell them we can do this. example is 'jdtls' for java. Suggested fix:
```
diff --git a/runtime/lua/vim/lsp/protocol.lua b/runtime/lua/vim/lsp/protocol.lua
index ececc41cee..72397bad8e 100644
--- a/runtime/lua/vim/lsp/protocol.lua
+++ b/runtime/lua/vim/lsp/protocol.lua
@@ -459,6 +459,7 @@ function protocol.make_client_capabilities()
             properties = {
               'additionalTextEdits',
               'command',
+              'documentation',
             },
           },
           tagSupport = {
```